File: python/problem040.py
index = 1
product_position = 1
total_length = 0
result = 1

while True:
    for i in str(index):
        total_length += 1
        if total_length == product_position:
            result *= int(i)
            product_position *= 10
    if product_position == 10000000:
        break
    index += 1

print(result)

File: javascript/problem040.js
var index = 1;
var productPosition = 1;
var totalLength = 0;
var result = 1;

while(productPosition<10000000){
  strIndex = index.toString();
  for(i = 0; i<strIndex.length; i++){
    totalLength = totalLength + 1;
    if(totalLength == productPosition){
      result = result * parseInt(strIndex.charAt(i));
      productPosition = productPosition * 10;
    }
  }
  index = index + 1;
}

console.log(result);
